### ISSUE TITLE
Error loading tribes/initialization/atlanteans/images/flag_00_pc.png: Unknown pixel format on sdl2_image 2.6.0

### ISSUE THREAD

**Issue by svenstaro** [widelands/widelands#5469]
Widelands 1.0 crashes when compiled and ran against sdl2_image 2.6.0. Startup says this:

```
❯ widelands
This is Widelands Version 1.0 (Release)
[00:00:00.000 real] INFO: Set home directory: /home/svenstaro/.local/share/widelands
[00:00:00.000 real] INFO: Set configuration file: /home/svenstaro/.config/widelands/config
[00:00:00.000 real] INFO: Adding directory: /usr/share/widelands
[00:00:00.000 real] INFO: selected language: (system language)
[00:00:00.000 real] INFO: using locale en_US.UTF-8
[00:00:00.001 real] INFO: Byte order: little-endian
[00:00:00.055 real] INFO: Graphics: Try to set Videomode 1144x263
[00:00:00.169 real] INFO: Graphics: OpenGL: Version "4.6.0 NVIDIA 515.57"
[00:00:00.169 real] INFO: Graphics: SDL_GL_RED_SIZE is 8
[00:00:00.169 real] INFO: Graphics: SDL_GL_GREEN_SIZE is 8
[00:00:00.169 real] INFO: Graphics: SDL_GL_BLUE_SIZE is 8
[00:00:00.169 real] INFO: Graphics: SDL_GL_ALPHA_SIZE is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_BUFFER_SIZE is 24
[00:00:00.169 real] INFO: Graphics: SDL_GL_DOUBLEBUFFER is 1
[00:00:00.169 real] INFO: Graphics: SDL_GL_DEPTH_SIZE is 24
[00:00:00.169 real] INFO: Graphics: SDL_GL_STENCIL_SIZE is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_ACCUM_RED_SIZE is 16
[00:00:00.169 real] INFO: Graphics: SDL_GL_ACCUM_GREEN_SIZE is 16
[00:00:00.169 real] INFO: Graphics: SDL_GL_ACCUM_BLUE_SIZE is 16
[00:00:00.169 real] INFO: Graphics: SDL_GL_ACCUM_ALPHA_SIZE is 16
[00:00:00.169 real] INFO: Graphics: SDL_GL_STEREO is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_MULTISAMPLEBUFFERS is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_MULTISAMPLESAMPLES is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_ACCELERATED_VISUAL is 1
[00:00:00.169 real] INFO: Graphics: SDL_GL_CONTEXT_MAJOR_VERSION is 2
[00:00:00.169 real] INFO: Graphics: SDL_GL_CONTEXT_MINOR_VERSION is 1
[00:00:00.169 real] INFO: Graphics: SDL_GL_CONTEXT_FLAGS is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_CONTEXT_PROFILE_MASK is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_SHARE_WITH_CURRENT_CONTEXT is 0
[00:00:00.169 real] INFO: Graphics: SDL_GL_FRAMEBUFFER_SRGB_CAPABLE is 0
[00:00:00.169 real] INFO: Graphics: OpenGL: Double buffering enabled
[00:00:00.169 real] INFO: Graphics: OpenGL: Max texture size: 32768
[00:00:00.169 real] INFO: Graphics: OpenGL: ShadingLanguage: "4.60 NVIDIA"
[00:00:00.171 real] INFO: **** GRAPHICS REPORT ****
[00:00:00.171 real] INFO: VIDEO DRIVER GLVND x11
[00:00:00.171 real] INFO: Display #0: 2560x1440 @ 144hz SDL_PIXELFORMAT_RGB888
[00:00:00.171 real] INFO: Display #1: 1440x2560 @ 144hz SDL_PIXELFORMAT_RGB888

Caught exception (of type '17ImageLoadingError') in outermost handler!
The exception said: [/build/widelands/src/widelands-1.0/src/graphic/image_io.h:41] Error loading tribes/initialization/atlanteans/images/flag_00_pc.png: Unknown pixel format

This should not happen. Please file a bug report on version 1.0(Release).
and remember to specify your operating system.
```

This is on Arch Linux with gcc 12.

It works fine when I go back down to sdl2_image 2.0.5.

sdl2_image 2.6.0 is compiled with these options:
```
gcc -DPACKAGE_NAME=\"SDL2_image\" -DPACKAGE_TARNAME=\"SDL2_image\" -DPACKAGE_VERSION=\"2.6.0\" -DPACKAGE_STRING=\"SDL2_image\ 2.6.0\" -DPACKAGE_BUGREPORT=\"https://github.com/libsdl-org/SDL_image/issues\" -DPACKAGE_URL=\"\" -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DSDL_BUILD_MAJOR_VERSION=2 -DSDL_BUILD_MINOR_VERSION=6 -DSDL_BUILD_MICRO_VERSION=0 -DPACKAGE=\"SDL2_image\" -DVERSION=\"2.6.0\" -DUSE_STBIMAGE=1 -DLOAD_AVIF=1 -DLOAD_JPG=1 -DLOAD_JXL=1 -DLOAD_PNG=1 -DLOAD_TIF=1 -DLOAD_WEBP=1 -DLOAD_BMP=1 -DLOAD_GIF=1 -DLOAD_LBM=1 -DLOAD_PCX=1 -DLOAD_PNM=1 -DLOAD_SVG=1 -DLOAD_TGA=1 -DLOAD_XCF=1 -DLOAD_XPM=1 -DLOAD_XV=1 -DLOAD_QOI=1 -DLOAD_AVIF_DYNAMIC=\"libavif.so.14\" -DSDL_IMAGE_SAVE_PNG=1 -DSDL_IMAGE_SAVE_JPG=1 -I.     -DAVIF_DLL   -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -flto=auto -I/usr/include/SDL2 -D_REENTRANT  -Wall -fvisibility=hidden -MT showimage.o -MD -MP -MF .deps/showimage.Tpo -c -o showimage.o showimage.c
```
so it definitely has PNG support.

EDIT: So it appears sdl2_image defaults to stb-image. See [CHANGELOG](https://github.com/libsdl-org/SDL_image/releases/tag/release-2.6.0). When providing `--disable-stb-image` it works just fine. I suppose we could consider this bug fixed but it might be something you want to look at.

**Comment by klaus-halfmann** [comment]
Do we have a Bug in  sdl2_image  2.0.6?

**Comment by Noordfrees** [comment]
Possibly related? https://github.com/libsdl-org/SDL_image/issues/294

**Comment by Noordfrees** [comment]
SDL_image released a point release 2.6.1 to fix this bug. Can someone please check whether the bug is gone with 2.6.1 and if so close this issue?

**Comment by klaus-halfmann** [comment]
Homebrew just Upgraded to: sdl2_image--2.6.0_1  now master starts up again, will palyn some testgame tommorow but I think we cal close this. Or do wen need another test using Linux?
